pennylane/devices/tf_ops.py

```diff
@@ -191,3 +191,108 @@ def CRot(a, b, c):
         :math:`|0\rangle\langle 0|\otimes \mathbb{I}+|1\rangle\langle 1|\otimes R(a,b,c)`
     """
     return tf.linalg.diag(CRZ(c)) @ (CRY(b) @ tf.linalg.diag(CRZ(a)))
+
+
+def DoubleExcitation(phi):
+    r"""
+    Double excitation rotation.
+    Args:
+        phi (float): rotation angle
+    Returns:
+        tf.Tensor[float]: Double excitation rotation matrix

```

Comment:
As pointed out by Josh in the previous PR ``phi = tf.cast(phi, dtype=C_DTYPE)``, ``phi`` being cast as complex.
```suggestion
        tf.Tensor[complex]: Double excitation rotation matrix

```